test/boilerplate_suggestors/advanced_props_and_state_class_migrator_test.dart

```diff
@@ -70,610 +255,1387 @@ void main() {
+            //   1. Check on the boilerplate migration status of the library it comes from.
+            //   2. Once the library has released a version that includes updated boilerplate,
+            //      bump the lower bound of your dependency to that version in your `pubspec.yaml`, and run `pub get`.
+            //   3. Re-run the migration script with the following flag:
+            //      pub run over_react_codemod:boilerplate_upgrade --convert-classes-with-external-superclasses
+            //   4. Once the migration is complete, you should notice that $externalSuperclassName has been deprecated. 
+            //      Follow the deprecation instructions to consume the replacement by either updating your usage to
+            //      the new class name and/or updating to a different entrypoint that exports the version(s) of 
+            //      $externalSuperclassName that is compatible with the new over_react boilerplate.
+            class $propsClassName extends $externalSuperclassName {
               String foo;
               int bar;
             }
-    
-            @Component2()
-            class BarComponent extends UiStatefulComponent2<BarProps, BarState> {
-              @override
-              render() {
-                return Dom.ul()(
-                  Dom.li()('Foo: ', props.foo),
-                  Dom.li()('Bar: ', props.bar),
-                );
+            
+            $componentDecl
+          ''';
+
+        group('but does add a FIXME comment', () {
+          setUp(() {
+            // When it is run the first time, nothing should happen since
+            // we don't know if the custom classes are external or not.
+            testSuggestor()(expectedPatchCount: 0, input: input);
+            testSuggestor()(
+              expectedPatchCount: 1,
+              input: input,
+              expectedOutput: expectedOutputWithExternalSuperclassReasonComment,
+            );
+          });
+
+          test('', () {
+            expect(
+                converter.visitedNames,
+                {
+                  publicPropsClassName: null,
+                },
+                reason:
+                    '$publicPropsClassName should not be converted since $externalSuperclassName is external, '
+                    'and the --convert-classes-with-external-superclasses flag is not set');
+          });
+
+          test(
+              'which then gets removed from the declaration that is converted to a mixin, and replaced '
+              'with updated instructions on the new concrete class declaration when the script '
+              'is re-ran with the --convert-classes-with-external-superclasses flag set',
+              () {
+            // Run it a third time - this time simulating `--convert-classes-with-external-superclasses`
+            // being set - which allows conversion of external superclasses
+            testSuggestor(convertClassesWithExternalSuperclass: true)(
```

Comment:
One of the examples of the second run of the codemod as mentioned in the comment above.
```suggestion
            testSuggestor()(expectedPatchCount: 0, input: expectedOutputWithExternalSuperclassReasonComment);
            testSuggestor(convertClassesWithExternalSuperclass: true)(
```

Reply:
@sydneyjodon-wk the testSuggestor is already being run twice before this in the `setUp` of the test.

Reply:
Yes, that's why I'm confused as to why it isn't run twice here as well. In `boilerplate_upgrade.dart`, it looks like every time the codemod is run, AdvancedPropsAndStateClassMigrator is run twice. The first time not considering the flag and the second time with the flag. 

It seems like the test setup is running the codemod once (so running `AdvancedPropsAndStateClassMigrator` twice like you said). So I’m just wondering why in this case when simulating running the codemod again, `AdvancedPropsAndStateClassMigrator` is only run once?

Reply:
But I guess the tests are still passing without the added run because `converter.visitedNames` isn't set back to empty in tearDown. I just didn’t know if it would be more realistic to run the suggester twice for each run of the codemod.